src/Nox.Solution/NoxSolution.cs

```diff
@@ -1,15 +1,80 @@
 using FluentValidation;
 using Nox.Solution.Validation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Nox.Solution;
 
 public class NoxSolution : Solution
 {
+    private List<EntityRelationshipWithType> _relationshipsToCreate = new List<EntityRelationshipWithType>();
+    private readonly object _relationshipsToCreateLock = new object();
+
     public string? RootYamlFile { get; internal set; }
 
     internal void Validate()
     {
         var validator = new SolutionValidator();
         validator.ValidateAndThrow(this);
     }
+
+    public List<EntityRelationshipWithType> GetRelationshipsToCreate(Func<string, Type?> getTypeByNameFunc)
```

Comment:
Decrease complexity, compute only relations for a given entity, and avoid having state